src/options/template.rs

```diff
@@ -1,16 +1,41 @@
-use std::fmt;
-use std::fmt::Display;
+use std::fmt::{self, Display};
+use std::io::Write;
 
-use saucer::{anyhow, clap, Parser, Result};
+use console::Term;
+use dialoguer::Select;
+use saucer::{anyhow, clap, Parser, Utf8PathBuf, ValueEnum};
 use serde::{Deserialize, Serialize};
 
+use crate::{error::RoverError, Result};
+
 #[derive(Debug, Clone, Serialize, Deserialize, Parser)]
 pub struct TemplateOpt {
     /// Filter templates by the available language
     #[clap(long = "language", value_enum)]
     pub language: Option<ProjectLanguage>,
 }
 
+impl TemplateOpt {
+    pub fn prompt_language(&self) -> Result<ProjectLanguage> {
```

Comment:
```suggestion
    pub fn get_or_prompt_language(&self) -> Result<ProjectLanguage> {
```
I think that better indicates that it won't _always_ prompt

Reply:
that makes sense to me!